components/selectors/matching.rs

```diff
@@ -724,6 +724,13 @@ fn matches_simple_selector<E, F>(
             flags_setter(element, HAS_EMPTY_SELECTOR);
             element.is_empty()
         }
+        Component::Scope => {
+            if let Some(ref scope_element) = context.shared.scope_element {
```

Comment:
I'd just use `match` instead of `if let Some(..) { } else {}`, but your call.

Reply:
Hmm, maybe a good idea.